modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientExperimentalCodegen.java

```diff
@@ -82,6 +82,9 @@ public PythonClientExperimentalCodegen() {
 
         apiTemplateFiles.remove("api.mustache");
         apiTemplateFiles.put("python-experimental/api.mustache", ".py");
+        if (hasHttpSignatureMethods(this.fullAuthMethods)) {
```

Comment:
Per my debugging in InteliJ fullAuthmethods doesn't exist yet.
this.OpenAPI (the spec does)
How about using this?
```
        Map<String, SecurityScheme> securitySchemeMap = openAPI.getComponents() != null ? openAPI.getComponents().getSecuritySchemes() : null;
        List<CodegenSecurity> authMethods = config.fromSecurity(securitySchemeMap);
        if (hasHttpSignatureMethods(authMethods)) {
            apiTemplateFiles.put("python-experimental/signing.mustache", ".py");
        }
```

Reply:
For the record I don't know when we should use supportingFiles.add vs apiTemplateFiles.put
It looks like supportingFiles.add is used more for conditional statments, so maybe we should
- use `supportingFiles.add(..`
- move the above if statement into which uses `supportingFiles.add(..` into the `processOpts` method
What do you think?

Reply:
yes, that's what I'm trying to do. For some reason the compiler cannot find the config symbol. I'll keep looking.

Reply:
Ah config s where you are. Just delete it. It should be `fromSecurity(securitySchemeMap)`

Reply:
oh I just realized there are two distinct classes, DefaultGenerator and DefaultCodeGen...
PythonClientExperimentalCodegen extends from DefaultCodeGen, not DefaultGenerator, and config is defined in DefaultGenerator.

Reply:
Sounds good

Reply:
It looks like putting
hasHttpSignatureMethods
in modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ProcessUtils.java
would work because both classes can use it.

Reply:
I have a basic question. When the constructor is invoked, has the OAS spec been parsed? Isn't it done later? Because I'm seeing a NPE and wonder if this is because the OAS spec hasn't been parsed yet, hence codegen wouldn't know what the security schemes are in the constructor.

Reply:
The spec is parsed, I am seeing the above code successfully generate authMethods:
```
authMethods = {ArrayList@2705}  size = 5
 0 = {CodegenSecurity@2714} "CodegenSecurity{name='api_key', type='apiKey', scheme='null', hasMore=true, isBasic=false, isOAuth=false, isApiKey=true, isBasicBasic=false, isBasicBearer=false, bearerFormat='null', vendorExtensions={}, keyParamName='api_key', isKeyInQuery=false, isKeyInHeader=true, isKeyInCookie=false, flow='null', authorizationUrl='null', tokenUrl='null', scopes=null, isCode=false, isPassword=false, isApplication=false, isImplicit=false}"
 1 = {CodegenSecurity@2715} "CodegenSecurity{name='api_key_query', type='apiKey', scheme='null', hasMore=true, isBasic=false, isOAuth=false, isApiKey=true, isBasicBasic=false, isBasicBearer=false, bearerFormat='null', vendorExtensions={}, keyParamName='api_key_query', isKeyInQuery=true, isKeyInHeader=false, isKeyInCookie=false, flow='null', authorizationUrl='null', tokenUrl='null', scopes=null, isCode=false, isPassword=false, isApplication=false, isImplicit=false}"
 2 = {CodegenSecurity@2716} "CodegenSecurity{name='bearer_test', type='http', scheme='bearer', hasMore=true, isBasic=true, isOAuth=false, isApiKey=false, isBasicBasic=false, isBasicBearer=true, bearerFormat='JWT', vendorExtensions={}, keyParamName='null', isKeyInQuery=false, isKeyInHeader=false, isKeyInCookie=false, flow='null', authorizationUrl='null', tokenUrl='null', scopes=null, isCode=false, isPassword=false, isApplication=false, isImplicit=false}"
 3 = {CodegenSecurity@2717} "CodegenSecurity{name='http_basic_test', type='http', scheme='basic', hasMore=true, isBasic=true, isOAuth=false, isApiKey=false, isBasicBasic=true, isBasicBearer=false, bearerFormat='null', vendorExtensions={}, keyParamName='null', isKeyInQuery=false, isKeyInHeader=false, isKeyInCookie=false, flow='null', authorizationUrl='null', tokenUrl='null', scopes=null, isCode=false, isPassword=false, isApplication=false, isImplicit=false}"
 4 = {CodegenSecurity@2718} "CodegenSecurity{name='petstore_auth', type='oauth2', scheme='null', hasMore=false, isBasic=false, isOAuth=true, isApiKey=false, isBasicBasic=false, isBasicBearer=false, bearerFormat='null', vendorExtensions={}, keyParamName='null', isKeyInQuery=false, isKeyInHeader=false, isKeyInCookie=false, flow='implicit', authorizationUrl='http://petstore.swagger.io/api/oauth/dialog', tokenUrl='null', scopes=[{scope=write:pets, hasMore=true, description=modify pets in your account}, {scope=read:pets, hasMore=null, description=read your pets}], isCode=false, isPassword=false, isApplication=false, isImplicit=true}"
```
Are you invoking it inside processOpts?

Reply:
> The spec is parsed, I am seeing the above code successfully generate authMethods:
> 
> ```
> authMethods = {ArrayList@2705}  size = 5
>  0 = {CodegenSecurity@2714} "CodegenSecurity{name='api_key', type='apiKey', scheme='null', hasMore=true, isBasic=false, isOAuth=false, isApiKey=true, isBasicBasic=false, isBasicBearer=false, bearerFormat='null', vendorExtensions={}, keyParamName='api_key', isKeyInQuery=false, isKeyInHeader=true, isKeyInCookie=false, flow='null', authorizationUrl='null', tokenUrl='null', scopes=null, isCode=false, isPassword=false, isApplication=false, isImplicit=false}"
>  1 = {CodegenSecurity@2715} "CodegenSecurity{name='api_key_query', type='apiKey', scheme='null', hasMore=true, isBasic=false, isOAuth=false, isApiKey=true, isBasicBasic=false, isBasicBearer=false, bearerFormat='null', vendorExtensions={}, keyParamName='api_key_query', isKeyInQuery=true, isKeyInHeader=false, isKeyInCookie=false, flow='null', authorizationUrl='null', tokenUrl='null', scopes=null, isCode=false, isPassword=false, isApplication=false, isImplicit=false}"
>  2 = {CodegenSecurity@2716} "CodegenSecurity{name='bearer_test', type='http', scheme='bearer', hasMore=true, isBasic=true, isOAuth=false, isApiKey=false, isBasicBasic=false, isBasicBearer=true, bearerFormat='JWT', vendorExtensions={}, keyParamName='null', isKeyInQuery=false, isKeyInHeader=false, isKeyInCookie=false, flow='null', authorizationUrl='null', tokenUrl='null', scopes=null, isCode=false, isPassword=false, isApplication=false, isImplicit=false}"
>  3 = {CodegenSecurity@2717} "CodegenSecurity{name='http_basic_test', type='http', scheme='basic', hasMore=true, isBasic=true, isOAuth=false, isApiKey=false, isBasicBasic=true, isBasicBearer=false, bearerFormat='null', vendorExtensions={}, keyParamName='null', isKeyInQuery=false, isKeyInHeader=false, isKeyInCookie=false, flow='null', authorizationUrl='null', tokenUrl='null', scopes=null, isCode=false, isPassword=false, isApplication=false, isImplicit=false}"
>  4 = {CodegenSecurity@2718} "CodegenSecurity{name='petstore_auth', type='oauth2', scheme='null', hasMore=false, isBasic=false, isOAuth=true, isApiKey=false, isBasicBasic=false, isBasicBearer=false, bearerFormat='null', vendorExtensions={}, keyParamName='null', isKeyInQuery=false, isKeyInHeader=false, isKeyInCookie=false, flow='implicit', authorizationUrl='http://petstore.swagger.io/api/oauth/dialog', tokenUrl='null', scopes=[{scope=write:pets, hasMore=true, description=modify pets in your account}, {scope=read:pets, hasMore=null, description=read your pets}], isCode=false, isPassword=false, isApplication=false, isImplicit=true}"
> ```
> 
> Are you invoking it inside processOpts?

no, from the constructor. There are a few test cases when the "openAPI" field is null, so I have to test for that too.

Reply:
Can you mark this as resolved?